ValheimPlus/PlayerStats.cs

```diff
@@ -29,89 +29,111 @@ private static void Postfix(Player __instance)
-                    weaponType = "Unarmed";
-
-                if(weaponType != "Unarmed") 
+                if (__instance.m_character.IsPlayer())
                 {
-                    try
+                    ItemDrop.ItemData item = __instance.m_character.GetRightItem();
+                    string weaponType = "";
+
+                    if (item == null)
+                    {
+                        weaponType = "Unarmed";
+                    }
+                    else if (item.IsWeapon())
                     {
-                        weaponType = __instance.GetRightItem().m_shared.m_skillType.ToString();
+                        try
+                        {
+                            weaponType = item.m_shared.m_skillType.ToString();
+                        }
+                        catch (System.Exception e) { }
                     }
-                    catch (System.Exception e)
-                    { }
 
-                    isHoe = (__instance.GetRightItem().m_shared.m_name == "$item_hoe" ? true : false);
-                    isHammer = (__instance.GetRightItem().m_shared.m_name == "$item_hammer" ? true : false);
-                }
-                
-                if(weaponType != "")
-                switch (weaponType)
-                {
-                    case "Swords":
-                        v = v - (v * (Configuration.Current.StaminaUsage.swords) / 100);
-                        break;
-                    case "Knives":
-                        v = v - (v * (Configuration.Current.StaminaUsage.knives) / 100);
-                        break;
-                    case "Clubs":
-                        v = v - (v * (Configuration.Current.StaminaUsage.clubs) / 100);
-                        break;
-                    case "Polearms":
-                        v = v - (v * (Configuration.Current.StaminaUsage.polearms) / 100);
-                        break;
-                    case "Spears":
-                        v = v - (v * (Configuration.Current.StaminaUsage.spears) / 100);
-                        break;
-                    case "Axes":
-                        v = v - (v * (Configuration.Current.StaminaUsage.axes) / 100);
-                            break;
-                    case "Bows":
-                        v = v - (v * (Configuration.Current.StaminaUsage.bows) / 100);
-                        break;
-                    case "Unarmed":
-                        v = v - (v * (Configuration.Current.StaminaUsage.unarmed) / 100);
-                        break;
-                    case "Pickaxes":
-                        v = v - (v * (Configuration.Current.StaminaUsage.pickaxes) / 100);
-                        break;
-                    default:
-                        break;
+                    if (weaponType != "")
+                    {
+                        switch (weaponType)
+                        {
+                            case "Swords":
```

Comment:
Can we use the `Skills.SkillType` enum for comparisons here for better future base game compatibility?

Reply:
Done.